### PR TITLE
counting duplicates 

### PR DESCRIPTION
**Pull request by aepaysinger**

### CODE REVIEW

File: src/challenges/counting_duplicate.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  text = text.toLowerCase()
+  let charCounter = {}
+  let total = 0
+  for (const x of text) {
+    if (x in charCounter) {
+      charCounter[x] += 1
+    } else {
+      charCounter[x] = 1
+    }
+  }
```
Comment (W-Ely) [inline src/challenges/counting_duplicate.js:11]: Use text.split().reduce(() instead.

File: src/challenges/counting_duplicate.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  for (const x in charCounter) {
+    if (charCounter[x] >= 2) {
+      total += 1
+    }
+  }
```
Comment (W-Ely) [inline src/challenges/counting_duplicate.js:16]: Same, for practice always try to avoid for loops.

try a obj.keys().reduce() here.

**Review by W-Ely — CHANGES_REQUESTED** [review]
See comments.

File: src/challenges/counting_duplicate.js (hunk 1 above)
Comment (aepaysinger) [reply src/challenges/counting_duplicate.js:11]: i understand that text.split() will turn the text into an array but i don't understand how to use text.split().reduce()
i struggled a bit building the object to begin with.

Comment (W-Ely) [reply src/challenges/counting_duplicate.js:11]: [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) is an array method that takes 2 arguments, a function for the first and an initial value for the second.  The function needs to take 2 values, the first is the accumulator (or the thing you are constructing's initial value) and the second currentValue (a single item from the list)

```javaScript
let counts = text.split("").reduce(
  (charCounter, char) => 
    if (char in charCounter) {
      charCounter[char] += 1
    } else {
      charCounter[char] = 1
    }
    return charCounter
  },
  {}
)
```

See if the above works for you and then try to convert the one below too.  It isn't about this way being _better_, it's about developing a feel and understanding of all the builtin methods.

Comment (aepaysinger) [reply src/challenges/counting_duplicate.js:11]: why the {} at the end?

Comment (W-Ely) [reply src/challenges/counting_duplicate.js:11]: > [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) is an array method that takes 2 arguments, a function for the first **and an initial value for the second.**

`{}` is the starting value for `charCounter`

**Comment by aepaysinger** [comment]
i used key in the second one because for some reason that is what it took for me to understand what was happening, but is key okay to use there?

**Review by W-Ely — APPROVED** [review]

**Pull request merged**